### PR TITLE
Fix RatioWeighting (DensityWeighting) manipulator

### LINKED ISSUE #1758: RatioWeightingImpl.hpp deprecated
`picongpu/src/picongpu/include/particles/manipulators/RatioWeightingImpl.hpp` seems to be deprecated.

```C++
const float_X densityRatioDes =
    traits::GetDensityRatio<T_DesParticle>::type::getDefaultValue();
const float_X densityRatioSrc =
    traits::GetDensityRatio<T_SrcParticle>::type::getDefaultValue();
```

A scan of all the code returned only those two results for the method `getDefaultValue()` which is also not defined anywhere anymore.

### PR DESCRIPTION
**Pull request by PrometheusPi**
This pull request fixes compile errors when using `RatioWeightingImpl` during species initialization as manipulator. 

- [x] It is **still required to add** `#include "particles/manipulators/RatioWeightingImpl.hpp"` in `speciesInitialization.param`. @ax3l and @psychocoderHPC is this intended? -> fixed

Thanks @n01r for finding this bug and helping to fix it. 

This closes #1758 and also (the wrongly defined) #1757. 

***

@ax3l's notes:
- only the first two commits are relevant for a backport

### CODE REVIEW

**Comment by ax3l** [comment]
Dang, we refactored the trait and the implementation at the same time when bringing that feature to mainline in #1018 (and should have added a test).
(Sorry, the example I gave you yesterday was in my local branch side-hacking that feature before mainline where it still worked ^^)

The actual bug then came in via #956 (ratio was always 1.0, #955) and survived/became active during #1018.

In #1018, after several refactorings, I think I just RT checked the proton-weighting manipulator.

File: src/picongpu/include/particles/manipulators/RatioWeightingImpl.hpp [hunk 1]
```diff
@@ -73,9 +73,9 @@ struct RatioWeightingImpl
         if (isDesParticle && isSrcParticle)
         {
             const float_X densityRatioDes =
-                traits::GetDensityRatio<T_DesParticle>::type::getDefaultValue();
+	      traits::GetDensityRatio<T_DesParticle>::type::getValue();
             const float_X densityRatioSrc =
-                traits::GetDensityRatio<T_SrcParticle>::type::getDefaultValue();
+	      traits::GetDensityRatio<T_SrcParticle>::type::getValue();
```
Comment (ax3l) [inline src/picongpu/include/particles/manipulators/RatioWeightingImpl.hpp]: both lines: introduced in an untested combination of #956 and #1018 ...

**Comment by ax3l** [comment]
> It is still required to add #include "particles/manipulators/RatioWeightingImpl.hpp" in speciesInitialization.param. [...] is this intended?

weird, `speciesInitialization` is included after `particleConfig` in `_defaultParam.loader` and there it is. No, you should not need to include the manipulator in `speciesInitialization`...

File: src/picongpu/include/particles/traits/GetDensityRatio.hpp [hunk 2]
```diff
@@ -25,6 +25,8 @@
 #include "traits/Resolve.hpp"
 
 #include <boost/mpl/if.hpp>
+#include "particles/memory/frames/Frame.hpp"
+
 
 namespace picongpu
```
Comment (ax3l) [inline src/picongpu/include/particles/traits/GetDensityRatio.hpp]: @PrometheusPi can you please also correct the missed alignment in line 60?

File: src/picongpu/include/particles/manipulators/RatioWeightingImpl.hpp [hunk 3]
```diff
@@ -73,9 +73,9 @@ struct RatioWeightingImpl
         if (isDesParticle && isSrcParticle)
         {
             const float_X densityRatioDes =
-                traits::GetDensityRatio<T_DesParticle>::type::getDefaultValue();
+                traits::GetDensityRatio<T_DesParticle>::type::getValue();
```
Comment (ax3l) [inline src/picongpu/include/particles/manipulators/RatioWeightingImpl.hpp]: ah no, your change is invalid. we use `value_identifier` in `speciesDefinition` which is a "default" value for an identifier that *could* be an l value, but is not used as such in the species.

**Comment by ax3l** [comment]
I'll push the following changes:
- `RatioWeighting`: had inconsistent implementation in `.hpp` (with `Impl`) -> no parameters -> no `Impl` interface needed
- `ProtonTimesWeightingImpl` (works): rename without `Impl` (since no parameters, too)

**Comment by ax3l** [comment]
@psychocoderHPC wuhu, approved. feel free to merge then :)

File: src/picongpu/include/particles/manipulators/RatioWeightingImpl.def [hunk 4]
```diff
@@ -40,7 +40,7 @@ namespace manipulators
  * note: needs the densityRatio flag on both species,
  *       used by the GetDensityRatio trait.
  */
-struct RatioWeighting;
+struct RatioWeightingImpl;
```
Comment (PrometheusPi) [inline src/picongpu/include/particles/manipulators/RatioWeightingImpl.def]: @ax3l Was this the reason why the extra include was needed?

**Pull request merged**